src/drivers/dw/ssi-spi.c

```diff
@@ -151,7 +151,8 @@ struct spi {
 #define spi_reg_write(spi, reg, val) \
 			io_reg_write(spi->plat_data->base + reg, val)
 
-extern struct ipc *_ipc;
+/* deprecated - will be passed by callers, used by IPC only */
+struct sof *_sof;
```

Comment:
I would consider renaming the whole struct to something else e.g. sof_context or system_context etc.
Preceding local instance with underscore makes it hard to read and analyze the code.

Reply:
yep, sounds fine to me. I will get back to this now that v1.3 is out.